kloppy/domain/services/transformers/dataset.py

```diff
@@ -236,6 +245,15 @@ def __change_point_coordinate_system(self, point: Union[Point, Point3D]):
         else:
             return Point(x=x, y=y)
 
+    def __change_attacking_direction(self, record: DataRecord):
+        new_attacking_direction = self._to_orientation.get_attacking_direction(
+            record.period
+        )
+        period = replace(
```

Comment:
I'm not totally sure a new instance of a `Period` should be created. My first idea would be that if this is needed, the `periods` attribute in the metadata must be replaced.

Reply:
Each `DataRecord` of a dataset and the `Metadata` have a `Period` attribute with (potentially) an attacking direction. When the `Orientation` of a dataset is transformed, I change all these attacking directions such that they are consistent with the new `Orientation`. 

The attacking direction should not have a function anymore elsewhere in the code. Hence, it is not strictly necessary but I think it makes sense to keep it consistent everywhere.